features/step_definitions/machine.rb

```diff
@@ -90,3 +90,22 @@
     end
   }
 end
+
+Given /^I run the steps #{NUMBER} times or exit when co machine-api is degraded:$/ do |num, steps_string|
```

Comment:
Sorry, I don't understand this step, and the eval_regex

Reply:
I basically reused this code from existing , this is to get exit is the degraded status is found before the 40th iteration .Passing the steps_string on regex.